zerver/lib/streams.py

```diff
@@ -917,12 +914,8 @@ def stream_to_dict(
         # passing stream data to spectators.
         stream_weekly_traffic = None
 
-    if setting_groups_dict is not None:
-        can_remove_subscribers_group = setting_groups_dict[stream.can_remove_subscribers_group_id]
-    else:
-        can_remove_subscribers_group = get_group_setting_value_for_api(
-            stream.can_remove_subscribers_group
-        )
```

Comment:
This block is removed because now all callers of `stream_to_dict` pass the `setting_groups_dict` argument, so it's never `None`.

I didn't remove `None` from the type annotation for `setting_groups_dict`, because it would requires to change the interface of `stream_to_dict` callers and also the interface of those callers.

Reply:
Ok. But we should update `setting_groups_dict` type in `stream_to_dict` in a follow-up.